### PR TITLE
tech(migrate to ui-kit): Make nav based on styled-components

### PR DESCRIPTION
**Pull request by jorgelainfiesta**
## Purpose

Using the same technologies and approaches across the projects allows us to build on top of the work we've already done and to deliver consistent experiences. As part of that project, this PR starts moving the styles in Zeus to styles based on `ui-kit`.

## Approach

We do not have the setup to transform `ui-kit` into a formal independent addon. The discussion about where the place for each module should be is still happening. However, we need to switch away from other techniques for styling because they will be harder to correct as we make progress in Zeus.

For that reason, I introduce `faux-ui-kit` in this PR. It is a provisional solution that allows us to consume what we need now. It can be easily replaced once things are prepared for the change to actually using `ui-kit`. 

I started with the Navigation because it seemed to be what was using more styles.

## Next Steps
- Replace all the alternative styling forms in use (classes, SASS, etc) for `styled-components` and `styled-system`
- Remove the SASS and any other Gatsby styles from the app.

## Screenshots

Screen Shot 2019-09-16 at 5 06 49 PM

Screen Shot 2019-09-16 at 5 07 04 PM

### CODE REVIEW

File: src/components/Layout.js [hunk 1]
```diff
@@ -1,59 +1,68 @@
...
 import { Flex } from 'rebass';
-
+import ThemeProvider from '../faux-ui-kit/ThemeProvider';
+// import styled from 'styled-components';
 import './all.sass';
 
+// const Logo = styled.h1`
+//   font-size: 16px;
+//   color: red;
+// `;
```
Comment (jgnieuwhof) [inline src/components/Layout.js]: 🔥 the comments ?

File: src/components/Layout.js [hunk 2]
```diff
@@ -1,59 +1,68 @@
...
+
 const TemplateWrapper = ({ children }) => {
   const { title, description } = useSiteMetadata();
   return (
-    <Flex flexDirection='column' height='100%'>
-      <Helmet>
-        <html lang='en' />
-        <title>{title}</title>
-        <meta name='description' content={description} />
+    <ThemeProvider>
```
Comment (jgnieuwhof) [inline src/components/Layout.js]: Seems like Gatsby provides a bit of a pattern to wrap the application with elements like providers: https://www.gatsbyjs.org/docs/browser-apis/#wrapRootElement.

Something like the following in `gatsby-browser.js`: 
```
const React = require("react")
const ThemeProvider  = require('./src/faux-ui-kit/ThemeProvider');

exports.wrapRootElement = ({ element }) => {
  return (
    <ThemeProvider>
      {element}
    </ThemeProvider>
  )
}
```

File: src/components/LargeNav.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      {gatewayNodes.map(({ slug, name, id }) => (
+        <NavLink to={`/gateways/${slug}`} key={id}>
+          {name}
+        </NavLink>
+      ))}
+      <NavLink to={`/editor`}>
+        Editor
+      </NavLink>
+      <IconLink
+        href='https://github.com/netlify-templates/gatsby-starter-netlify-cms'
```
Comment (jgnieuwhof) [inline src/components/LargeNav.js]: While we're here we can probably burn this link away :fire:.

File: src/components/MediumNav.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      <MenuLinks>
+        {gatewayNodes.map(({ slug, name, id }) => (
+          <MediumMenuItem to={`/gateways/${slug}`} key={id}>
+            {name}
+          </MediumMenuItem>
+        ))}
+        <MediumMenuItem to='/editor'>
+          Editor
+        </MediumMenuItem>
+        <MediumMenuItem
```
Comment (jgnieuwhof) [inline src/components/MediumNav.js]: Same w/ this one :fire:.

File: src/components/LargeNav.js (hunk 3 above)
Comment (taras) [reply src/components/LargeNav.js]: We should change it to point to the zeus repo. @jorgelainfiesta can you please add a link to the repo to `package.json` and then us the link from package in this file.

**Comment by taras** [comment]
> Would it be worth pulling in the actual ui-kit via package.json from the ui-kit src for now?

I tried doing this but ui-kit is not ready. @jorgelainfiesta and I discussed using it but there is very little there that can help us. I think we can align on practices of ui-kit - like using `styled-components` but ui-kit is not yet ready for us to consume in this project.

**Comment by taras** [comment]
Oops... sorry, I accidentally closed the PR

File: src/components/Layout.js (hunk 2 above)
Comment (jorgelainfiesta) [reply src/components/Layout.js]: I'm not familiar with Gatsby, but loading the theme in `gatsby-browser.js` would make the static pages generated to have wrong styling? I understand that `gastby-browser` is ran on the last step after pages have been generated already. Perhaps I'm not correctly understanding how we're usnig Gatsby.

File: src/components/LargeNav.js (hunk 3 above)
Comment (jorgelainfiesta) [reply src/components/LargeNav.js]: Yes sure!

**Comment by jorgelainfiesta** [comment]
> If answer to ^^ is no, we could use aliases to import our faux-ui-kit components as if they were actual ui-kit components

I'll ping you so you can give me some guidance on how to do this the right way

> Is it intentional to define two completely separate menus for different size screens?

I think I was following the pattern used in `pro-portal`, but indeed in this case we could just style them differently. The only issue I find is that with `styled-components` the styles code look pretty crumbled with media queries everywhere. 

After our conversation of today, I'll propose a new nav that works better for this project.

However you raised a good point, perhaps we shouldn't even invest into creating a mobile menu because it's unlikely that this app is interested in having a mobile version.

**Comment by taras** [comment]
> However you raised a good point, perhaps we shouldn't even invest into creating a mobile menu because it's unlikely that this app is interested in having a mobile version.

Yeah, it's difficult to see how this project can be used on mobile.

**Pull request closed without merging**